nepalingo-web/src/pages/TestYourself.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect, useRef } from "react";
+import useDictionary from "@/hooks/useDictionary";
+import { generate } from "random-words";
+import Header from "@/components/header/Header";
+import Button from "@/components/Button";
+import { useStreak } from "@/hooks/StreakContext";
+import { useLanguage } from "@/hooks/Langauge";
+import { getNextWord } from "@/lib/getNextWord";
+
+const TestYourself: React.FC = () => {
+  const { updateStreak } = useStreak();
+  const { selectedLanguage } = useLanguage();
+  const [word, setWord] = useState("today");
+
+  const [options, setOptions] = useState<string[]>([
+    "today",
+    "rice",
+    "hot",
+    "salt",
+  ]);
+  const [selectedOption, setSelectedOption] = useState<string | null>(null);
+  const [isCorrect, setIsCorrect] = useState<boolean | null>(null);
+  const { data, isLoading } = useDictionary({
+    language: selectedLanguage || "newari",
+    word,
+  });
+  const wordGeneratorRef = useRef<ReturnType<typeof getNextWord> | null>(null);
+
+  useEffect(() => {
+    updateStreak(); // Trigger streak update on flashcard page load
+    wordGeneratorRef.current = getNextWord(selectedLanguage || "newari");
+  }, [selectedLanguage]);
+
+  const getOptions = (word: string) => {
+    const randomWords = generate({ exactly: 3 }) as string[];
+    randomWords.push(word);
+    const shuffledOptions = randomWords.sort(() => Math.random() - 0.5);
+    setOptions(shuffledOptions);
+  };
+
+  const handleNextQuestion = async () => {
+    const generator = await wordGeneratorRef.current;
+    if (generator) {
+      const nextWord = generator?.next()?.value;
+
+      if (typeof nextWord === "string") {
+        setWord(nextWord);
+        setSelectedOption(null);
+        setIsCorrect(null);
+        getOptions(nextWord);
+      } else {
+        console.error("Generated word is not a string.");
```

Comment:
Sorry, I don't understand. Does this generator only give string values, or can it also give non-string values? Please also handle those cases in the code if it can provide non-string values.

Reply:
I actually I have handled those cases in test-yourself and flashcards